console-subscriber/src/builder.rs

```diff
@@ -138,7 +138,7 @@ impl Builder {
     /// defaults.
     ///
     /// [environment variable]: `Builder::with_default_env`
-    pub fn server_addr(self, server_addr: impl Into<SocketAddr>) -> Self {
+    pub fn server_addr(self, server_addr: impl Into<ServerAddr>) -> Self {
```

Comment:
Open to other API design here. E.g., two separate, mutually-exclusive methods, one for TCP and one for Unix sockets. I went with the new `ServerAddr` type to avoid breaking existing users.

Reply:
Hmm, this approach seems fine to me if we want to make UDS and TCP mutually exclusive. 

However, it occurs to me that we _might_ want to add a separate method for setting a UDS server address, but _not_ making them mutually exclusive. That way, we would allow the user to configure serving the console API on both a TCP port _and_ a Unix socket, if desired. Do you think this is something people will want to be able to do?

If we did have separate methods, it might be worth adding a new method `tcp_server_addr` (or something), and deprecating the existing `server_addr` in favor of that, to make it clearer when called what type of socket is being configured.

On the other hand, the current approach is probably a nicer API surface if we don't want to allow serving over both TCP and UDS...

Reply:
alternatively, if we wanted to support this, we could just make `server_addr` called multiple times append addresses to a list.

In the meantime, though, I think we should update the documentation for `server_addr` to make sure that it's very clear that multiple calls will overwrite any previously set values.

Reply:
Makes sense! Done.